### PR TITLE
Include ExplicitInterfaceImplementation method in Inheritance Margin 

### PR DESCRIPTION
**Pull request by Cosifne**
Fix https://github.com/dotnet/roslyn/issues/53667

### CODE REVIEW

File: src/EditorFeatures/Test/InheritanceMargin/InheritanceMarginTests.cs [hunk 1]
```diff
@@ -989,6 +989,65 @@ public class {{|target1:Bar2|}} : IBar{lessThanToken}int{greaterThanToken}, IBar
                 itemForFooInBar2);
         }
 
+        [Fact]
+        public Task TestCSharpExplicitInterfaceImplementation()
+        {
+            var lessThanToken = SecurityElement.Escape("<");
+            var greaterThanToken = SecurityElement.Escape(">");
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Test/InheritanceMargin/InheritanceMarginTests.cs:996]: what the heck is this?

File: src/EditorFeatures/Test/InheritanceMargin/InheritanceMarginTests.cs [hunk 2]
```diff
@@ -989,6 +989,65 @@ public class {{|target1:Bar2|}} : IBar{lessThanToken}int{greaterThanToken}, IBar
...
+            var markup = $@"
+interface {{|target2:IBar|}}{lessThanToken}T{greaterThanToken}
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Test/InheritanceMargin/InheritanceMarginTests.cs:998]: i legit have no idea what's going on.  why is this not just `interface {{|target2:IBar|}}<T>`?

**Review by CyrusNajmabadi — APPROVED** [review]
Approve... but tests are wacky.

File: src/EditorFeatures/Test/InheritanceMargin/InheritanceMarginTests.cs (hunk 2 above)
Comment (davidwengier) [reply src/EditorFeatures/Test/InheritanceMargin/InheritanceMarginTests.cs:998]: I'm guessing this is eventually used in an XML definition for a project. Personally I'd be fine with just `interface {{|target2:IBar|}}&lt;T&gt;`

**Review by davidwengier — APPROVED** [review]
Thank you for fixing my issue :)

File: src/EditorFeatures/Test/InheritanceMargin/InheritanceMarginTests.cs (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/EditorFeatures/Test/InheritanceMargin/InheritanceMarginTests.cs:998]: yeah.  if this is eventually used in an xml definition, it shoudl be *that* code that escapes this code :)

File: src/EditorFeatures/Test/InheritanceMargin/InheritanceMarginTests.cs (hunk 1 above)
Comment (Cosifne) [reply src/EditorFeatures/Test/InheritanceMargin/InheritanceMarginTests.cs:996]: That's the escaped character for '<' and '>' for xml.

File: src/EditorFeatures/Test/InheritanceMargin/InheritanceMarginTests.cs (hunk 2 above)
Comment (Cosifne) [reply src/EditorFeatures/Test/InheritanceMargin/InheritanceMarginTests.cs:998]: : ) Let me take a look and see if I can escape these char in the helper method

File: src/EditorFeatures/Test/InheritanceMargin/InheritanceMarginTests.cs (hunk 1 above)
Comment (Cosifne) [reply src/EditorFeatures/Test/InheritanceMargin/InheritanceMarginTests.cs:996]: Oops, l mean  `&lt;` and `&gt;`. Github also rendering these characters.

File: src/EditorFeatures/Core/InheritanceMargin/AbstractInheritanceMarginService.cs [hunk 3]
```diff
@@ -92,8 +92,8 @@ private static bool CanHaveInheritanceTarget(ISymbol symbol)
                 return false;
             }
 
-            if (symbol is INamedTypeSymbol or IEventSymbol or IPropertySymbol ||
-                symbol.IsOrdinaryMethod())
+            if (symbol is INamedTypeSymbol or IEventSymbol or IPropertySymbol
+                or IMethodSymbol { MethodKind: MethodKind.Ordinary or MethodKind.ExplicitInterfaceImplementation })
```
Comment (Youssef1313) [inline src/EditorFeatures/Core/InheritanceMargin/AbstractInheritanceMarginService.cs:96]: Probably will need some tweaks for abstract statics (especially operators)

File: src/EditorFeatures/Test/InheritanceMargin/InheritanceMarginTests.cs [hunk 4]
```diff
@@ -138,12 +143,12 @@ private static void VerifyInheritanceTarget(TestInheritanceTargetItem expectedTa
     <Project Language=""{markup1.languageName}"" AssemblyName=""Assembly1"" CommonReferences=""true"">
         <ProjectReference>Assembly2</ProjectReference>
         <Document>
-            {markup1.markupInProject1}
+            {markup1.markupInProject1.Replace("<", s_lessThanToken).Replace(">", s_greaterThanToken)}
         </Document>
```
Comment (Youssef1313) [inline src/EditorFeatures/Test/InheritanceMargin/InheritanceMarginTests.cs:147]: Would CDATA work and avoids this logic completely?

File: src/EditorFeatures/Core/InheritanceMargin/AbstractInheritanceMarginService.cs (hunk 3 above)
Comment (Cosifne) [reply src/EditorFeatures/Core/InheritanceMargin/AbstractInheritanceMarginService.cs:96]: I am not fully tracking the compiler side work. But do you mean this https://github.com/dotnet/csharplang/issues/4436 ? I see it's still at design stage so probably it won't be taken into C# 11.  Is it already implemented?

Comment (Youssef1313) [reply src/EditorFeatures/Core/InheritanceMargin/AbstractInheritanceMarginService.cs:96]: @Cosfine Yes it's in https://github.com/dotnet/roslyn/tree/features/StaticAbstractMembersInInterfaces. Some IDE work (e.g, completion) is being done currently in the feature branch.

Comment (Cosifne) [reply src/EditorFeatures/Core/InheritanceMargin/AbstractInheritanceMarginService.cs:96]: Then I would prefer first merge main to the feature branch (which I suppose bot would do it regularly), then make change in that branch.
But thanks to let me know this. : ) I will track this change

**Pull request merged**